pkgs/applications/emulators/dosbox-x/default.nix

```diff
@@ -0,0 +1,87 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, alsa-lib
+, autoconf
+, automake
+, file
+, fluidsynth
+, freetype
+, glib
+, libpcap
+, libpng
+, libslirp
+, libXext
+, libxkbfile
+, libXrandr
+, nasm
+, ncurses
+, pkg-config
+, SDL2
+, which
+, yad
+, zlib
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "dosbox-x";
+  version = "2023.09.01";
+
+  src = fetchFromGitHub {
+    owner = "joncampbell123";
+    repo = "dosbox-x";
+    rev = "dosbox-x-v${finalAttrs.version}";
+    hash = "sha256-gEqIPXrq16p6VC9aGW2sRSWE0SG9hiMWeRzSoDvUbB0=";
+  };
+
+  nativeBuildInputs = [
+    autoconf
+    automake
+    file
+    pkg-config
+  ];
+
+  buildInputs = [
+    alsa-lib
+    fluidsynth
+    freetype
+    glib
+    libpcap
+    libpng
+    libslirp
+    libXext
+    libxkbfile
+    libXrandr
+    nasm
+    ncurses
+    SDL2
+    which
+    yad
+    zlib
+  ];
+
+  postPatch = ''
+    substituteInPlace vs/sdlnet/ltmain.sh vs/sdlnet/acinclude/libtool.m4 --replace /usr/bin/file ${file}/bin/file
+    patchShebangs .
+  '';
+
+  # dosbox-x really wants to use its own SDL_NET
+  # https://github.com/joncampbell123/dosbox-x/blob/5eea489cda7c1e6fbd1ded6932dd8459b5e0693a/build-sdl2#L42
+  preConfigure = ''
+    ./autogen.sh
+    pushd vs/sdlnet
+    ./build-dosbox.sh
+    popd
+  '';
```

Comment:
1. Instead of calling the custom autogen script, we can let automatic hooks take care of it by putting `autoreconfHook` into `nativeBuildInputs`.

2. This way of handling the vendored SDL_net doesn't sit well with me… At the very least it certainly breaks cross-compiling because this doesn't pass through all of the flags needed there.

Would this be maintainable for you? If not, and you're fine with co-maintaining, then you can also add me to maintainers - I'm interested in it for the NEC PC-98 system emulation. :slightly_smiling_face:
```suggestion
  # dosbox-x uses a modified SDL_net
  # https://github.com/joncampbell123/dosbox-x/blob/5eea489cda7c1e6fbd1ded6932dd8459b5e0693a/build-sdl2#L42
  preConfigure = let 
    custom-SDL_net = (SDL_net.override { SDL = SDL2; }).overrideAttrs (oa: {
      pname = "${finalAttrs.pname}-vendored-${oa.pname}";
      inherit (finalAttrs) version src;
      sourceRoot = "${finalAttrs.src.name}/vs/sdlnet";
      nativeBuildInputs = oa.nativeBuildInputs ++ [
        autoconf
        automake
      ];  
      preConfigure = ''
        ./autogen.sh
      '';
      configureFlags = oa.configureFlags ++ (with lib.strings; [
        (enableFeature true "static")
        (enableFeature false "shared")
      ]); 
    });
  in ''
    ln -s ${custom-SDL_net} vs/sdlnet/linux-host
  '';
```

Reply:
I think the autoreconfhook stage has resolved this issue but I'm not sure if you were getting at something meatier with the static options. Leaving open for now and you can resolve if all's well.